artifactory/utils/docker/buildinfo.go

```diff
@@ -236,15 +236,19 @@ func (builder *buildInfoBuilder) setBuildProperties() (int, error) {
 		log.Error("Fail to create new content writer for docker layer")
 		return 0, err
 	}
-	defer writer.Close()
-	for _, item := range builder.layers {
-		writer.Write(item)
-	}
+	writeLayers(builder.layers, writer)
 	reader := content.NewContentReader(writer.GetFilePath(), content.DefaultKey)
 	defer reader.Close()
 	return builder.serviceManager.SetProps(services.PropsParams{Reader: reader, Props: props})
 }
 
+func writeLayers(layers []utils.ResultItem, writer *content.ContentWriter) {
+	defer writer.Close()
```

Comment:
also it's discouraged to ignore the errors, causes bugs like the one we had :)
(think about failure when closing the file but the app still rely on the file being written)